idx/bigtable/bigtable.go

```diff
@@ -533,3 +539,38 @@ func (b *BigtableIdx) prune() {
 		}
 	}
 }
+
+// AddDefs adds defs to the index.
+func (b *BigtableIdx) AddDefs(defs []schema.MetricDefinition) {
+	pre := time.Now()
+
+	b.MemoryIndex.AddDefs(defs)
+
+	if b.cfg.UpdateBigtableIdx {
+		// Blocking write to make sure all get enqueued
+		for _, def := range defs {
+			b.writeQueue <- writeReq{recvTime: time.Now(), def: &def}
+		}
+	}
+
+	statControlRestoreDuration.Value(time.Since(pre))
+}
+
+// DeleteDefs deletes the matching key.
+func (b *BigtableIdx) DeleteDefs(defs []schema.MetricDefinition, archive bool) {
+	pre := time.Now()
+
+	b.MemoryIndex.DeleteDefs(defs, archive)
+
+	if b.cfg.UpdateBigtableIdx {
+		// TODO - this could create a lot of go routines if many deletes come in at the same time
+		// Maybe better to enhance the write queue to process these?
+		go func() {
+			for _, def := range defs {
+				b.deleteDef(&def)
```

Comment:
comparing this to b.deleteDefs() (which deals with `[]idx.Archive`) makes me realize that we should log an error when deleteDef returns one.